src/components/AddUsersModal/index.jsx

```diff
@@ -0,0 +1,328 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm, SubmissionError } from 'redux-form';
+import { Button, Icon, Modal } from '@edx/paragon';
+
+import H3 from '../H3';
+import TextAreaAutoSize from '../TextAreaAutoSize';
+import StatusAlert from '../StatusAlert';
+import FileInput from '../FileInput';
+import { validateEmailTemplateFields, validateEmailAddresses } from '../../utils';
+
+class AddUsersModal extends React.Component {
```

Comment:
I think we can divide this modal into three react components
Status,
EmailSection,
TemplateSection